src/evict/evict_lru.c

```diff
@@ -574,52 +558,70 @@ __evict_pass(WT_SESSION_IMPL *session)
 		 * workers, it must service the urgent queue in case all
 		 * application threads are busy.
 		 */
-		if (cache->evict_empty_score < WT_EVICT_EMPTY_SCORE_CUTOFF ||
+		if (cache->evict_empty_score < WT_EVICT_SCORE_CUTOFF ||
 		    (!WT_EVICT_HAS_WORKERS(session) &&
 		    !__evict_queue_empty(cache->evict_urgent_queue)))
 			WT_RET(__evict_lru_pages(session, true));
 
+		if (cache->pass_intr != 0)
```

Comment:
Your change doesn't really change how pass_intr is being used, I am suspicious that it could be disruptive to the eviction server when handles are closed after a checkpoint has finished, since I suspect that checkpoint handles will be discarded by sweep via ``__wt_conn_dhandle_discard_single``, which will interrupt what the eviction server is doing.